[nicleejy] iP

nicleejy opened the pull request:
# Introducing UwU_TaskMaster

> "Y-Y-Youw mind is fow having i-ideas, nyot howding them. UwU" – David Awwen ([source](https://dansilvestre.com/productivity-quotes))

UwU_TaskMastew fwees youw mind of having t-to wemembew things you nyeed t-to do. It's,
- text-based
- easy t-to weawn
- ~~FAST~~ *runs away* SUPEW FAST t-to use

aww >w< you nyeed t-to do is,

1. downwoad i-it fwom hewe.
2. doubwe-cwick it.
3. add youw tasks.
4. wet i-it manyage youw tasks fow you 😉

And it is **FWEE**!

Features:
- [ ] Managing to-dos
- [ ] Managing deadlines 
- [ ] Managing events
- [ ] Reminders (coming soon)
---
If you Java pwogwammew, you c-can use i-it t-to pwactice Java too. *runs away* Hewe's the x3 main method for the CLI version of UwU_TaskMaster:

```java
/**
     * Runs the bot and reads in user input in a loop, parses it and responds accordingly
     * @param args program args
     */
    public static void main(String[] args) {

        System.out.println("Hewwo! I'm UwU_TaskMaster! How c-can I hewp you?!?");

        Scanner scanner = new Scanner(System.in);
        TaskFileReaderWriter taskReaderWriter = new TaskFileReaderWriter();


        TaskManager taskManager = taskReaderWriter.loadDataFromFile();

        if (!taskReaderWriter.createTaskFile()) {
            System.out.println("Error creating data file");
            return;
        }

        Parser parser = new Parser(taskManager);
        String input;

        while (true) {

            input = scanner.nextLine();

            if (input.contains("bye")) {

                if (!taskReaderWriter.updateTaskFile(taskManager)) {
                    System.out.println("Error updating data file");
                    return;
                }
                System.out.println(Format.formatResponse(Response.BYE_BYE.toString()));
                break;
            }

            String output = parser.processInput(input);

            System.out.println(Format.formatResponse(output));
        }
        scanner.close();
    }
```

Commit:
Add Gradle support

nicleejy committed:
Add echo functionality for Duke

nicleejy committed:
change chat layout

nicleejy committed:
added storage functionality

nicleejy committed:
add mark and unmark task functions, abstracted tasks as class

nicleejy committed:
add support for deadline, event and todo

nicleejy committed:
added testcase for bot

nicleejy committed:
update exception classes with javadoc

nicleejy committed:
added deletion function

nicleejy committed:
add javadoc comments and create enum for responses, code cleanup

nicleejy committed:
clean up code, abstracted task list, add enums for responses

nicleejy committed:
more code refactoring, update test cases

nicleejy committed:
Add functionality to save tasks to hard disk

nicleejy committed:
Add javadoc comments

nicleejy committed:
Add formatting for date and time, store dates as LocalDateTime object

nicleejy committed:
Merge branch 'branch-Level-7'
Merge task file changes to master

nicleejy committed:
Merge branch 'branch-Level-8'
Merge date time processing feature into master

nicleejy committed:
Abstracted functionality to different classes

nicleejy committed:
Merge branch 'add-gradle-support'
add gradle file wrapper to project

nicleejy committed:
add gradle wrapper

nicleejy committed:
add junit test cases

nicleejy committed:
write javadoc for updated code

nicleejy committed:
fix code to match coding style defined in checkstyle.xml

nicleejy committed:
Add support for finding tasks by keyword

nicleejy committed:
Merge branch 'branch-Level-9'
Merge find feature into master branch

nicleejy committed:
Merge branch 'branch-A-JavaDoc'
Merge JavaDoc changes to master branch

nicleejy committed:
Fix merge conflict, introduce coding standard updates to master

nicleejy committed:
Add more javadoc fix checkstyle violations

ruiyigan commented on src/main/java/data/TaskFileReaderWriter.java:
```diff
@@ -0,0 +1,163 @@
+     *
+     * @return boolean value indicating the success of the file creation.
+     */
+    public boolean createTaskFile() {
```
Since this method returns a boolean, you may want to consider using a prefix such as `is`, `has`, `was`. Perhaps something like `canCreateTaskFile` or `hasCreatedTaskFile`?

ruiyigan commented on src/main/java/data/TaskFileReaderWriter.java:
```diff
@@ -0,0 +1,163 @@
+ * The TaskFileReaderWriter class provides methods for reading and writing tasks to and from a file.
+ * It includes methods for loading tasks from a file and updating a task file with tasks from a
+ * TaskManager object, and creating a new task file.
+ * @author Nicholas Lee
```
I like that you included `@author` in your comments.

ruiyigan commented on src/main/java/data/TaskFileReaderWriter.java:
```diff
@@ -0,0 +1,163 @@
+     * @param taskManager The TaskManager object containing the tasks to be written to the file.
+     * @return boolean value indicating the success of the file update operation.
+     */
+    public boolean updateTaskFile(TaskManager taskManager) {
```
Similar comment as the one above

ruiyigan commented on src/main/java/ui/Response.java:
```diff
@@ -0,0 +1,119 @@
+/**
+ * Various responses of UwU_TaskMaster
+ */
+public enum Response {
```
This is a good addition! It offers an easier way to manage responses. 👍

ruiyigan commented on userData/test.txt:
```diff
@@ -0,0 +1,2 @@
+Deadline|false|eat food|23/12/2023 1200
```
Just a suggestion. Perhaps you want to include test.txt in your .gitignore file to prevent the constant file change when committing changes.

ruiyigan reviewed (commented):
Overall, the coding standard is good! 👍 It follows the Java coding standard closely. There was only a minor issue I found regarding the naming of boolean methods. 

Please let me know if you have any queries regarding my comments.

nicleejy committed:
add GUI to bot

nicleejy committed:
Merge branch 'branch-Level-10'

nicleejy committed:
add assertions

nicleejy committed:
refactor code and fix code quality

nicleejy committed:
Merge branch 'branch-A-Assertions'
Merge assertion updates into master

nicleejy committed:
Merge branch 'branch-A-CodeQuality'

nicleejy committed:
Add query for free time slots in user schedule

Users were previously unable to query for free times in their list of
tasks.

To resolve this issue, adding a new feature allowing users to state a
duration and the bot automatically finds a free slot if it exists.

Introduce FreeTimeBlock class to represent free timeslots with start and
end timings.

Implement getFreeTimes() method in the TaskManager class to find free
slots based on the user's schedule.

A different String representation of FreeTimeBlock is generated depending on
whether it is bounded or unbounded.

This update allows for more efficient time management, and improves user
experience.

nicleejy committed:
Merge pull request #1 from nicleejy/branch-BCD-Extension

Add query for free time slots in user schedule

nicleejy committed:
convert ui to fxml

nicleejy committed:
finalise ui

nicleejy committed:
fix checkstyle violations and remove old ui files

nicleejy committed:
reset user text file

nicleejy committed:
Update README.md

nicleejy committed:
add Ui.png

nicleejy committed:
Merge branch 'master' of https://github.com/nicleejy/ip

nicleejy committed:
update readme images

nicleejy committed:
Update README.md

nicleejy committed:
add config.yml

nicleejy committed:
fix images in readme

nicleejy committed:
add instruction, change file pathing

nicleejy committed:
fix code quality

nicleejy committed:
shorten method length in TaskFileReaderWriter

nicleejy committed:
shorten method length in TaskManager

nicleejy committed:
prevent adding start later than end event

nicleejy committed:
fix bug in findFreeTimes feature causing out of bounds exception
